examples/nodejs/src/marlowe-object-flow.ts

```diff
@@ -0,0 +1,659 @@
+    stakeAddress: rewardAddress,
+    minimumLovelaceUTxODeposit: 3_000_000,
+    version: "v1",
+  });
+  const signedCborHex = await lifecycle.wallet.signTx(unsignedTx.tx.cborHex);
+  await lifecycle.restClient.submitContract(
+    unsignedTx.contractId,
+    transactionWitnessSetTextEnvelope(signedCborHex)
+  );
+  const txId = contractIdToTxId(unsignedTx.contractId);
+  return [unsignedTx.contractId, txId];
+  //----------------
+}
+
+type ValidationResults = "InvalidTags" | "InvalidContract" | DelayPaymentScheme;
+
+/**
+ * This function checks if the contract with the given id is an instance of the delay payment contract
+ * @param lifecycle
+ * @param contractId
+ * @returns
+ */
+async function validateExistingContract(
+  lifecycle: RuntimeLifecycle,
+  contractId: ContractId
+): Promise<ValidationResults> {
+  // First we try to fetch the contract details and the required tags
+  const contractDetails =
+    await lifecycle.restClient.getContractById(contractId);
+
+  const scheme = extractSchemeFromTags(contractDetails.tags);
+
+  if (!scheme) {
+    return "InvalidTags";
+  }
+
+  // If the contract seems to be an instance of the contract we want (meanin, we were able
+  // to retrieve the contract scheme) we check that the actual initial contract has the same
+  // sources.
+  // This has 2 purposes:
+  //   1. Make sure we are interacting with the expected contract
+  //   2. Share the same sources between different Runtimes.
+  //      When a contract source is uploaded to the runtime, it merkleizes the source code,
+  //      but it doesn't share those sources with other runtime instances. One option would be
+  //      to download the sources from the initial runtime and share those with another runtime.
+  //      Or this option which doesn't require runtime to runtime communication, and just requires
+  //      the dapp to be able to recreate the same sources.
+  const contractBundle = mkDelayPayment(scheme);
+  const { contractSourceId } =
+    await lifecycle.restClient.createContractSources(contractBundle);
+  const initialContract = await lifecycle.restClient.getContractSourceById({
+    contractSourceId,
+  });
+
+  if (!deepEqual(initialContract, contractDetails.initialContract)) {
+    return "InvalidContract";
+  }
+  return scheme;
+}
+
+async function main() {
+  const args = parseCli();
+  const config = await readConfig(args["--config"]);
+  const lucid = await Lucid.new(
+    new Blockfrost(config.blockfrostUrl, config.blockfrostProjectId),
+    config.network
+  );
+  lucid.selectWalletFromSeed(config.seedPhrase);
+  const rewardAddressStr = await lucid.wallet.rewardAddress();
+  const rewardAddress = rewardAddressStr
+    ? stakeAddressBech32(rewardAddressStr)
+    : undefined;
+  const runtimeURL = config.runtimeURL;
+
+  const wallet = mkLucidWallet(lucid);
+
+  const lifecycle = mkRuntimeLifecycle({
+    runtimeURL,
+    wallet,
+  });
+  await mainLoop(lifecycle, rewardAddress);
```

Comment:
Add a `try catch` that allows to display a detailed explanation of the error thrown : e.g 
```ts
  try{
    await mainLoop(lifecycle, rewardAddress);
  } catch (e) {
    console.log(`Error : ${JSON.stringify(e,null,4)}`);
  }
```
I had an issue, and I wasn't able to take action without modifying the codebase

Reply:
ok